### PR TITLE
Impl: Protected Routes and Logout for Test

### PR DESCRIPTION
**Pull request by brentmillen**
## Ticket/s

- [Route Protection](https://trello.com/c/CcSXsuHr/113-route-protection)

## Changes

- Implement protected routes
- Implement global auth provider for components
- Add unauthorized view
- Redirect user to login if no token
- updated merchant link _/my-products_
- Implement mock logout for test

## Screenshot

image

## How to test?

- checkout to branch
- cd apps/web
- run pnpm install
- cd ../.. (to root)
- navigate to different views as a logged in user (both merchant and consumer)
- navigate to views with no user logged in
- with access token, restart browser and navigate to views
- while on protected route, clear browser token in local storage
  - Go to dev tools
  - Application tab > Local storage
  - Select ey-commerce-token and clear manually
  
note: newer implementations are not part of this PR(ex. User Update view)

### CODE REVIEW

File: apps/web/src/views/AuthProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  ReactNode,
+  useEffect,
+} from 'react';
+import { decodeJwt, JWTPayload } from 'jose';
+
+interface AuthContextType {
+  user: any | null;
+  isLoading: boolean;
+  //   login: (user: User) => void;
+  //   logout: () => void;
```
Comment (markjoshuacarmen) [inline apps/web/src/views/AuthProvider.tsx]: please remove this comment

**Review by markjoshuacarmen — COMMENTED** [review]
please see comment

**Review by sandrex-SGV — APPROVED** [review]
LGTM!

**Review by TreshAnn — APPROVED** [review]
LGTM!

**Review by SGVGen — APPROVED** [review]
LGTM!

**Pull request merged**